app/src/main/java/io/github/omisie11/spacexfollower/util/ViewExtensions.kt

```diff
@@ -10,3 +23,41 @@ fun View.toggleVisibility(): View {
     }
     return this
 }
+/**
+ * Retrieve a color from the current [android.content.res.Resources.Theme].
+ */
+@ColorInt
+@SuppressLint("Recycle")
+fun Context.themeColor(
+    @AttrRes themeAttrId: Int
+): Int {
+    return obtainStyledAttributes(
+        intArrayOf(themeAttrId)
+    ).use {
+        it.getColor(0, Color.MAGENTA)
+    }
+}
+
+/**
+ * Retrieve a style from the current [android.content.res.Resources.Theme].
+ */
+@StyleRes
+fun Context.themeStyle(@AttrRes attr: Int): Int {
+    val tv = TypedValue()
+    theme.resolveAttribute(attr, tv, true)
+    return tv.data
+}
+
+@SuppressLint("Recycle")
+fun Context.themeInterpolator(@AttrRes attr: Int): Interpolator {
+    return AnimationUtils.loadInterpolator(
+        this,
+        obtainStyledAttributes(intArrayOf(attr)).use {
+            it.getResourceId(0, android.R.interpolator.fast_out_slow_in)
+        }
+    )
+}
+
+fun Context.getDrawableOrNull(@DrawableRes id: Int?): Drawable? {
+    return if (id == null || id == 0) null else AppCompatResources.getDrawable(this, id)
+}
```

Comment:
Could You extract those Extensions into separate file?
This file is named `ViewExtensions`, and only extensions on View classes should be placed here.
You could add for example file like "ResourcesExtensions" or something like that 😃

Reply:
okay, I will extract it in a separate file.